### PR TITLE
ENT-5038/master: Added 3 retries when starting daemons on AIX

### PR DESCRIPTION
**Pull request by nickanderson**
After reboot, sometimes, some daemons do not start successfully. This change adds up to 3 retries when starting daemons on AIX with a 5-second pause between retries.

Here is example output run while netcat is keeping port 5308 busy and preventing cf-serverd from starting:

```
#+begin_example
l0790p002_pub[/] > /etc/rc.d/init.d/cfengine3 start
-n Starting cf-execd...

-n Starting cf-serverd...
   error: Unable to start server
   re-try 1 starting /var/cfengine/bin/cf-serverd   error: Unable to start server
   re-try 2 starting /var/cfengine/bin/cf-serverd   error: Unable to start server
   re-try 3 starting /var/cfengine/bin/cf-serverd   error: Unable to start server

-n Starting cf-monitord...
#+end_example

```

### CODE REVIEW

**Comment by nickanderson** [comment]
```
#+begin_example
l0790p002_pub[/] > /etc/rc.d/init.d/cfengine3 start
-n Starting cf-execd...

-n Starting cf-serverd...
   error: Unable to start server
   re-try 1 starting /var/cfengine/bin/cf-serverd   error: Unable to start server
   re-try 2 starting /var/cfengine/bin/cf-serverd   error: Unable to start server
   re-try 3 starting /var/cfengine/bin/cf-serverd   error: Unable to start server

-n Starting cf-monitord...
#+end_example

```

File: misc/init.d/cfengine3.in [hunk 1]
```diff
@@ -312,7 +318,17 @@ cf_start_daemon()
     elif [ "$DEBIAN" = "1" ] && [ "$SSD" = "0" ]; then
         "$daemon" "$@"
         cf_touch "$LOCKDIR/$base_daemon"
-
+    elif [ "$AIX" = "1" ]; then
+        "$daemon" "$@" && break
```
Comment (vpodzime) [inline misc/init.d/cfengine3.in]: `break` what?

Comment (nickanderson) [reply misc/init.d/cfengine3.in]: ¯\_(ツ)_/¯

**Comment by nickanderson** [comment]
> Looks OK to me otherwise. We should make _cf-serverd_ tell that the port is occupied. Also, what are the reasons for other daemons not starting?

We don't know. There are no error messages and it's intermittent. By the time an operator logs on and runs the init script to restart the service, they seem to reliably start.

**Comment by nickanderson** [comment]
> We should make _cf-serverd_ tell that the port is occupied

I can change the log messages no problem, but I don't understand the difference between our existing error messages.

```
    int sd = OpenReceiverChannel(bind_address);

    if (sd == -1)
    {
        Log(LOG_LEVEL_ERR, "Unable to start server");
    }
    else if (listen(sd, queue_size) == -1)
    {
        Log(LOG_LEVEL_ERR, "listen failed. (listen: %s)", GetErrorStr());
        cf_closesocket(sd);
    }
```
 
I don't know the difference between `sd == -1` and `listen(sd, queue_size) == -1`. 
So, if you tell me what they should say, I can make the change, otherwise, do you want to tidy those error logs?

**Pull request closed without merging**